src/views/Gantt.vue

```diff
@@ -0,0 +1,243 @@
+import graphqlMixin from '@/mixins/graphql'
+import GanttChart from '@/components/cylc/gantt/GanttChart.vue'
+import DeltasCallback from '@/services/callbacks'
+import {
+  matchTasks,
+  platformOptions
+} from '@/components/cylc/analysis/filter'
+/** List of fields to request for each job */
+const jobFields = [
+  'name',
+  'id',
+  'submittedTime',
+  'startedTime',
+  'finishedTime',
+  'platform'
+]
+/** The query which retrieves historical Job timing statistics */
+const QUERY = gql`
+query ganttQuery ($workflows: [ID]) {
+  jobs(live: false, workflows: $workflows) {
+    ${jobFields.join('\n')}
+  }
+}
+`
+/** The callback which gets called when data comes in from the query */
+export class GanttCallback extends DeltasCallback {
+  /**
+   * @param {Object[]} jobs
+   */
+  constructor (jobs) {
+    super(jobs)
+    this.jobs = jobs
+  }
+  /**
+   * Add jobs contained in data to this.jobs
+   */
+
+  add (data) {
+    const taskNameList = Array.from(
+      new Set(data.jobs.map((job) => job.name))
+    )
+    this.uniqueTasks = Array.from(
+      new Set(data.jobs.map((job) => job.name))
+    )
+    const sortedData = Object.fromEntries(taskNameList.map(key => [key, []]))
+    for (let i = 0; i < data.jobs.length; i++) {
+      sortedData[data.jobs[i].name].push(data.jobs[i])
+    }
+    this.jobs.push(sortedData)
+  }
+  // called when new objects are added
+  // NOTE: we manually call this to add items which come through on the query
+
+  onAdded (added, store, errors) {
+    this.add(added)
+  }
+
+  // called when existing objects are updated
+  onUpdated (updated, store, errors) {
+    this.add(updated)
+  }
+}
+export default {
+  name: 'Gantt',
+  mixins: [
+    graphqlMixin
+  ],
+  components: {
+    GanttChart,
+  },
+  head () {
+    return {
+      title: getPageTitle('App.workflow', { name: this.workflowName })
+    }
+  },
+  beforeMount () {
+    this.jobsQuery()
+  },
+  data () {
+    const jobs = []
+    return {
+      tasksPerPage: 10,
+      callback: new GanttCallback(jobs),
+      /** Object containing all of the jobs added by the callback */
+      jobs,
+      timingOptions: [
+        { value: 'totalTimes', title: 'Total times' },
+        { value: 'runTimes', title: 'Run times' },
+        { value: 'queueTimes', title: 'Queue times' },
+      ],
+      jobsFilter: {
+        name: [],
+        timingOption: 'totalTimes',
+        platformOption: -1,
+      },
+    }
+  },
+  computed: {
+    // a list of the workflow IDs this view is "viewing"
+    // NOTE: we plan multi-workflow functionality so we are writing views
+    // to be mult-workflow compatible in advance of this feature arriving
+    workflowIDs () {
+      return [this.workflowId]
+    },
+    filteredJobs () {
+      if (this.jobsFilter.name.length === 0) {
+        return this.jobs
+      }
+      return [matchTasks(this.jobs[0], this.jobsFilter)]
+    },
+    platformOptions () {
+      return platformOptions(this.jobs)
```

Comment:
This function does not seem to be suited to the data format of `this.jobs`. The platform filtering is currently broken

Reply:
Ah yes I had forgotten to mention that. I had to reshape the data for the gantt chart (Which has broken the platform filtering)
Since we have the task selection box, do we need platform filtering? Or can we remove it?

Reply:
Up to you.

Reply:
I've fixed platform options. (and refactored filtering)